checkers/always_returning_checker.py

```diff
@@ -0,0 +1,103 @@
+            # Once there is not an explicit return, flag = false
+            flag = True
+            if not any([isinstance(child, astroid.Return) for child in (
+                    node.body + node.orelse)]):
+                flag = False
+            # set first_branch
+            if node.orelse and isinstance(node.orelse[0], astroid.If):
+                first_branch = node.orelse[0]
+            elif node.orelse:
+                for child in node.orelse:
+                    if isinstance(child, astroid.If):
+                        return self.helper_func(child, args)
+                # Return False if there is no return in else branch
+                if not any([isinstance(child, astroid.Return) for child in (
+                        node.orelse)]):
+                    flag = False
+                first_branch = "end"
+            else:
+                first_branch = "end"
+            # check if every branch has a explicit return
+            while first_branch != "end":
+                # deal with the condition of nested if
+                for child in first_branch.body:
+                    if isinstance(child, astroid.If):
+                        return self.helper_func(child, args)
+                # Return False if there is no return in this branch
+                if not any([isinstance(child, astroid.Return) for child in (
+                        first_branch.body + first_branch.orelse)]):
+                    flag = False
+                    first_branch = "end"
+                if first_branch.orelse and isinstance(first_branch.orelse[0],
+                                                      astroid.If):
+                    # deal with the condition of nested if
+                    for child in first_branch.orelse.body:
+                        if isinstance(child, astroid.If):
+                            return self.helper_func(child, args)
+                    first_branch = first_branch.orelse[0]
+                # Return False if there is no return in else branch
+                elif first_branch.orelse:
+                    for child in first_branch.orelse:
+                        if isinstance(child, astroid.If):
+                            return self.helper_func(child, args)
+                    if not any([isinstance(child, astroid.Return) for child in (
+                            first_branch.orelse)]):
+                        flag = False
+                    first_branch = "end"
+                else:
+                    first_branch = "end"
+            # check the flag and decide whether there is error
+            if flag:
+                self.add_message("always_returning_in_a_loop", node=node,
+                                 args=args)
+        # If it's a loop, check the body of the loop.
+        if isinstance(node, astroid.For) or isinstance(node, astroid.While):
+            for child in node.body:
+                self.helper_func(child, args=node.lineno)
```

Comment:
it would be visually nice if there was some more blank lines used between some code blocks in this function.